### ISSUE TITLE
Classification of sentence pair with two different languages

### ISSUE THREAD

**Issue by echan00** [google-research/bert#302]
I am hoping to classify sentence pairs in two different languages (English A and Chinese B). How would I go about this? 
Another possible better classification task I am considering is classifying triplet-sentences. E.g English A, Chinese B, and machine-translation of Chinese B in English.

OR
If the above is not recommended, I can also consider classifying English A and machine-translation of Chinese B in English, I feel this would be less data for the model to learn though.

**Comment by echan00** [comment]
@jacobdevlin-google I see you were part of multilingual, would absolutely appreciate your expertise here :)

**Comment by aditya-malte** [comment]
You could modify the MRPC class(which is also a pair classification task) in the BERT source code and fine tune on your dataset

**Comment by echan00** [comment]
@aditya-malte  yes, that is what I will try. It looks like using BERT-Base, Multilingual Cased version will be perfect for this.

What do you mean by this? Is it necessary to specify the language?
> Your input format would be something like:
> EngChinese

**Comment by aditya-malte** [comment]
Hello,
I meant to say that it could to be in the format:
Eng text {Sep} chinese text.
Thank you

**Comment by dengyuning** [comment]
> I am hoping to classify sentence pairs in two different languages (English A and Chinese B). Do I just use the multi-lingual BERT model?
> 
> Another possible classification task I am considering is classifying triplet-sentences. E.g English A, Chinese B, and machine-translation of Chinese B in English. Would this be worthwhile compared to just ONLY English A and Chinese B?
> 
> OR
> If the above is not recommended, I can also consider classifying English A and machine-translation of Chinese B in English, I feel this would be less data for the model to learn from though.

I have been working on a multilingual task recently and can I discuss it with you ?

**Comment by echan00** [comment]
Sure I've had good results

**Comment by aditya-malte** [comment]
@echan00 did you also perform Triplet classification. If yes, then how(because as far as I remember BERT was only trained on a next sentence prediction task, which is a sentence pair task)

Best,
Aditya Malte

**Comment by boxiangliu** [comment]
@echan00 I have also been working on multi-lingual sentence pair classification. Can I discuss this with you?

**Comment by echan00** [comment]
@aditya-malte  Sorry,  I didn't try any triplet classification.

@boxiangliu  Sure, erik.chan[at]gmail

**Comment by chiragsanghvi10** [comment]
Hi @echan00,
I have also been working on multi-lingual sentence similarity.

### for example:

> Sentence 1 (English)
> Sentence 2 (Translation in Hindi of Sentence 1)
> Sentence Similarity Score.
 
Any idea on how do I train for sentence similarity?

**Comment by echan00** [comment]
@chiragsanghvi10 this is what you're looking for, repurpose this https://github.com/google-research/bert/blob/master/README.md#sentence-and-sentence-pair-classification-tasks

**Comment by chiragsanghvi10** [comment]
Hi @echan00 , 

I am looking for fine-tuning multi-lingual model on

> Sentence 1 (Which is in English)

> Sentence 2 (Which is in Hindi, machine translation of Sentence 1)

> Similarity score of Sentence 1 and Sentence 2.
